frontend/app/src/App.tsx

```diff
@@ -910,11 +904,13 @@ export class App extends PureComponent<Props, State> {
     } else {
       // Set the title and favicon to their default values
       document.title = `${newPageName} · Streamlit`
-      handleFavicon(
-        `${process.env.PUBLIC_URL}/favicon.png`,
-        this.hostCommunicationMgr.sendMessageToHost,
-        this.endpoints
-      )
+      const favIcon = `${process.env.PUBLIC_URL}/favicon.png`
+      const imageUrl = getFaviconUrl(favIcon, this.endpoints)
+      this.hostCommunicationMgr.sendMessageToHost({
```

Comment:
We probably also want to send out this `SET_PAGE_FAVICON ` event here when `disableSetPageMetadata` is set, or? So this code might need some changes.

Reply:
Fixed. Added 
```
const favIcon = `${process.env.PUBLIC_URL}/favicon.png`
    const imageUrl = getFaviconUrl(favIcon, this.endpoints)
    this.hostCommunicationMgr.sendMessageToHost({
      type: "SET_PAGE_FAVICON",
      favicon: getFaviconUrl(imageUrl, this.endpoints),
    })
    ```
    right above.